### ISSUE TITLE
Lint/UnreachableLoop false positive when next appears in a branch

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop-hq/rubocop#8607]
In a loop where an `if` branch has both outcomes ending in `break` or `rescue`, but one branch has a `next` guard clause in it, Rubocop still complains about Lint/UnreachableLoop. Consider this:

```ruby
[1, 2].each do |x|
  p x
  if x == 1
    next if x == 1
    break
  else
    break
  end
end
```

This runs twice.

--------

## Expected behavior

Should not complain about Lint/UnreachableLoop

## Actual behavior

The loop runs twice, but Rubocop complains: "Lint/UnreachableLoop: This loop will have at most one iteration"

## Steps to reproduce the problem

Run sample code

## RuboCop version

```
0.89.1 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.4.5 x86_64-darwin19)
```

Love all the work you're doing 👍
